src/helpers/AIAssistant/Tools.ts

```diff
@@ -262,3 +262,59 @@ export const GetEhrNewsFeedPageTool = tool(
     })
   }
 );
+
+export const GraphingTool = tool(
+  async (input): Promise<string> => {
+
+    let response = await fetch('https://kwejahcwmsyzidk5vde5uyv33a0saruz.lambda-url.us-east-1.on.aws', {
```

Comment:
The assistant can generate invalid python. Here's an example. `timedelta` doesn't exist.
```
import matplotlib.pyplot as plt
from datetime import datetime
from collections import Counter

# Survey completion dates
survey_dates = [
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01",
    "2024-10-01"
]

# Count surveys per day
survey_count = Counter(survey_dates)

# Generate dates for the last 2 weeks
date_range = [datetime(2024, 9, 20) + timedelta(days=i) for i in range(14)]
date_labels = [date.strftime("%Y-%m-%d") for date in date_range]

# Prepare data for plotting
counts = [survey_count.get(date, 0) for date in date_labels]

# Plotting
plt.figure(figsize=(10, 5))
plt.bar(date_labels, counts, color='skyblue')
plt.xticks(rotation=45, ha='right')
plt.xlabel('Date')
plt.ylabel('Number of Surveys Completed')
plt.title('Surveys Completed Each Day Over the Last 2 Weeks')
plt.tight_layout()
plt.show()
```

500 Response:
```
{"error": "Error: name 'timedelta' is not defined"}
```

I THINK it needed to import this from the datetime lib (I'm not a Python person). e.g. `from datetime import datetime, timedelta`

Not to over-fixate on this specific failure...our code should account for mistakes from the AI Assistant. Kyle's relevant work: we check the result of Export Explorer Copilot's sql and automatically share errors back to the assistant and ask it to try again. And in any case, this tool needs an appropriate response when the fetch fails for any reason.